pyiron_base/jobs/job/extension/server/generic.py

```diff
@@ -486,6 +493,23 @@ def from_hdf(self, hdf, group_name=None):
         if "threads" in hdf_dict.keys():
             self._threads = hdf_dict["threads"]
         self._new_hdf = hdf_dict["new_h5"] == 1
+        options_used = [
+            "user",
+            "host",
+            "run_mode",
+            "queue",
+            "qid",
+            "structure_id",
+            "cores",
+            "run_time",
+            "memory_limit",
+            "accept_crash",
+            "threads",
+            "new_h5",
+        ]
+        for k, v in hdf_dict.items():
+            if k not in options_used:
+                self._additional_arguments[k] = v
```

Comment:
The `pysqa` version was just updated to `0.0.21` which now supports the `kwargs` so all I want the user to add is the arguments which are not yet handled by `pyiron_base`. As shown above examples could be `account` or `exclude`. 

Still I agree that the maintainability of this part is tricky. A more clean solution would be to move the whole storage of the server object to one central dictionary or ideally a `DataContainer` but that to me sounds like a larger change.

Reply:
I think I'm in favour of a broader but cleaner change? I understand it would take more time now, but with the implementation here, I worry about lost time in the future both in terms of stuff breaking/misbehaving and no one understanding how to fix it, and how much study will be required by future devs before they can make changes here.

At an absolute minimum we should break this list up into components with good variable names or even explicit comments about where these strings are coming from and why they should be forbidden.

Reply:
> Still I agree that the maintainability of this part is tricky. A more clean solution would be to move the whole storage of the server object to one central dictionary or ideally a `DataContainer` but that to me sounds like a larger change.

Not sure, how big of a change it actually would be, in #728 I did it for the `Executable` class which should be very similar.  Anyways, even without converting to `DataContainer` based storage, can't we just set a dictionary with the extra keys (which h5io should handle), i.e. have the new keys in an extra hierarchy level instead of a flat version?

Reply:
I am also fine with moving the additional arguments to a different dictionary.